clients/cobol-lsp-vscode-extension/src/services/CopybooksDownloader.ts

```diff
@@ -54,22 +56,25 @@ export class CopybooksDownloader implements vscode.Disposable {
             return;
         }
         const depDesc: DependenciesDesc = loadDepFile(depFileUri);
+
+        // TODO: REMOVE PROFILE OR ADD IT AFTER LOCAL CHECK
         const profile: string = await this.profileService.getProfile(depDesc.programName);
         if (!profile) {
             return;
         }
 
-        const missingCopybooks: string[] = await this.listMissingCopybooks(depDesc.copybooks, profile);
+        this.prioritizer.checkCopybooksPresentLocal(depDesc.copybooks,
+        vscode.workspace.getConfiguration(SETTINGS_SECTION).get(PATHS_LOCAL_KEY));
+        const notLocalCpy: string[] = this.prioritizer.getNotLocalCpy();
+        const localCpyExists: boolean = this.prioritizer.getLocalCpyURI().length > 0;
```

Comment:
Which is the value for ```Prioritizer``` to store internally the list of copybooks defined locally and not?
Can we simplify this logic just filtering from the copybooks list what is already present in local workspace and leave the others for the downloader component?

Reply:
Whoever needs the present or not present copybooks lists can access it from here.
What do you mean to leave for the downloader component? Downloader expects a list of copybooks to be downloaded

Reply:
if you expose the get() you need to avoid to work with old lists and this means that u have to invoke ```checkCopybooksPresentLocal()``` every time before get the list.
In my opinion the list should be generated on the fly every time.
The basic idea that I have in my mind is that copybook downloader needs to work with a list of copybooks, from this list we have to remove what is already present local, so I can imagine the prioritizer as a component that filter the copybooks list applying some criteria (for now check if is local)